### PR TITLE
[#4] [Chore] Set up the CD with Fastlane and GitHub Actions for iOS - part 2 of 2

### LINKED ISSUE #4: [Chore] Set up the CD with Fastlane and GitHub Actions for iOS
## Why

- No manual execution of releases for the iOS platform.

## Acceptance Criteria

- CD pipeline is run on GitHub Actions.
- CD pipeline run to build Staging and Production versions and upload to TestFlight.

### PR DESCRIPTION
**Pull request by Thieurom**
- Close #4
  - [x] ~Part 1: deploy iOS staging build to TestFlight~ (#40)
  - [x] Part 2: deploy iOS production build to TestFlight

## What happened 👀
- Add a new `fastlane`'s lane for building and uploading production app to Test Flight.
- Create workflow to build and deploy iOS production app.

## Insight 📝
`N/A`

## Proof Of Work 📹
- Workflows ran successfully on GitHub actions ([detail](https://github.com/nimblehq/flutter-ic-khanh-thieu/actions/runs/5736680527)):
  Screenshot 2023-08-02 at 16 16 53

- App's uploaded successfully on TestFight:
  Screenshot 2023-08-03 at 09 13 18

- Get invited:
   IMG_1927

-  Can install and run app on device:
  
  https://github.com/nimblehq/flutter-ic-khanh-thieu/assets/17972674/3ce0eaa6-1556-4737-803a-df9c8781d6cc

### CODE REVIEW

**Review by nkhanh44 — COMMENTED** [review]
Looks good, only one concern

File: .github/workflows/ios_deploy_production_to_testflight.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+name: iOS - Deploy Production build to TestFlight
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [ opened, synchronize, edited, reopened ]
+  # push:
+  #   branches:
+  #     - main
```
[the remark this answers is not in the document]
Comment (doannimble) [reply .github/workflows/ios_deploy_production_to_testflight.yml:8]: I agree with @nkhanh44 , we should trigger the production build only for `main` branch 🤔

**Review by doannimble — APPROVED** [review]
Apart from @nkhanh44 's comment. The rest lgtm

File: .github/workflows/ios_deploy_production_to_testflight.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Thieurom) [reply .github/workflows/ios_deploy_production_to_testflight.yml:8]: Oh, this was temporarily added to trigger GitHub action, then I forgot to revert it.
Will update it now 🙏

**Review by nkhanh44 — APPROVED** [review]

**Pull request merged**